engineering/docs/go_conventions.md

```diff
@@ -0,0 +1,139 @@
+
+## General
+
+### Logging
+
+Use `github.com/solo-io/go-utils/contextutils` for logging. 
+Initialize a context with useful information, including a version field.
+
+```go
+func getInitialContext() context.Context {
+	loggingContext := []interface{}{"version", version.Version}
+	ctx := contextutils.WithLogger(context.Background(), BotName)
+	ctx = contextutils.WithLoggerValues(ctx, loggingContext...)
+	return ctx
+}
+```
+
+Log using zap fields where possible. If a number of fields are added, break the line into a single
+ field per line: For instance:
+
+```go
+contextutils.LoggerFrom(ctx).Infow("Downloading repo archive",
+			zap.String("owner", r.owner),
+			zap.String("repo", r.repo),
+			zap.String("sha", r.sha))
+```
+
+### Interfaces
+
+Define interfaces when writing new utilities/modules that allow for mocking out dependencies and unit testing. 
+The default struct implementation for an interface should be have the same name, with a lowercase first letter. 
+A constructor should be used for instantiation. The interface, struct, and constructor should be near the top of a file. 
+
+```go
+type ChangelogReader interface {
+	GetChangelogForTag(ctx context.Context, tag string) (*Changelog, error)
+	ReadChangelogFile(ctx context.Context, path string) (*ChangelogFile, error)
+}
+
+type changelogReader struct {
+	code vfsutils.MountedRepo
+}
+
+func NewChangelogReader(code vfsutils.MountedRepo) ChangelogReader {
+	return &changelogReader{code: code}
+}
+
+func (c* changelogReader) GetChangelogForTag(ctx context.Context, tag string) (*Changelog, error) {
+	...
+}
+
+func (c* changelogReader) ReadChangelogFile(ctx context.Context, path string) (*ChangelogFile, error) {
+	...
+}
+```
+
+### Testing
+
+Unit tests should be written for every new interface function implementation or public function. 
+Mockgen should be used as necessary to mock out underlying implementation dependencies. 
+Unit tests should be written for every error and interesting edge case. 
+
+### Tracking changes
+
+Repos should prefer to use changelogs and solobot for issue/PR linking and release management. 
+See https://github.com/solo-io/go-utils/tree/master/changelogutils.
+
+## Errors
```

Comment:
As a general note for the`error` section, I think we should start using the `New` and `Is` functions that wrap the correspondent ones in the new `xerror` package in the standard library. See [here](https://github.com/solo-io/go-utils/blob/1c2dd400703ec1a7340609a6f5c00653ecf3d5c7/errors/utils.go#L23)

Reply:
This is done in `github.com/solo-io/go-utils/errors`, which I'm happy to add here. I would like to see examples of "Is" being useful, however, because I think it is very underwhelming. 

```
expected := changelogutils.TooManyChangelogFilesAddedError(2)
file, err := validator.ValidateChangelog(ctx)
Expect(file).To(BeNil())
// This fails
Expect(errors.Is(err, expected)).To(BeTrue())
// This succeeds
Expect(err.Error()).To(Equal(expected.Error()))
```